contrib/drivers/hpe/nimble/nimble.go

```diff
@@ -0,0 +1,346 @@
+		Description:      lun.Description,
+		AvailabilityZone: opt.GetAvailabilityZone(),
+		Metadata: map[string]string{
+			"Group":  lun.OwnedByGroup,
+			"Iqn":    lun.TargetName,
+			"LunId":  lun.Id,
+			"PoolId": poolId,
+		},
+	}, nil
+}
+
+func (d *Driver) DeleteVolume(opt *pb.DeleteVolumeOpts) error {
+	log.Infof("%v: Trying delete volume ...", DriverName)
+	poolId := opt.GetMetadata()["PoolId"]
+	//	err := d.client.DeleteVolume(poolId, lunId)
+	err := d.client.DeleteVolume(poolId, opt)
+	if err != nil {
+		log.Errorf("%v: delete volume failed, volume id =%s , Error:%s", DriverName, opt.GetId(), err)
+		return err
+	}
+	log.Infof("%v: remove volume success", DriverName)
+	return nil
+}
+
+func (d *Driver) ExtendVolume(opt *pb.ExtendVolumeOpts) (*model.VolumeSpec, error) {
+	log.Infof("%v: trying Extend volume...", DriverName)
+	poolId := opt.GetMetadata()["PoolId"]
+	_, err := d.client.ExtendVolume(poolId, opt)
+	if err != nil {
+		log.Errorf("%v: extend Volume Failed:", DriverName)
+		return nil, err
+	}
+
+	log.Infof("%v: extend volume success.", DriverName)
+	return &model.VolumeSpec{
+		BaseModel: &model.BaseModel{
+			Id: opt.GetId(),
+		},
+		Name:             opt.GetName(),
+		Size:             opt.GetSize(),
+		Description:      opt.GetDescription(),
+		AvailabilityZone: opt.GetAvailabilityZone(),
+	}, nil
+}
+
+func (d *Driver) CreateSnapshot(opt *pb.CreateVolumeSnapshotOpts) (*model.VolumeSnapshotSpec, error) {
+	log.Infof("%v: trying create snapshot...", DriverName)
+	poolId := opt.GetMetadata()["PoolId"]
+	snap, err := d.client.CreateSnapshot(poolId, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("%v: create snapshot success.", DriverName)
+	return &model.VolumeSnapshotSpec{
+		BaseModel: &model.BaseModel{
+			Id: opt.GetId(),
+		},
+		Name:        opt.GetName(),
+		Description: opt.GetDescription(),
+		VolumeId:    opt.GetVolumeId(),
+		Size:        0,
+		Metadata: map[string]string{
+			"SnapId": snap.Id,
+		},
+	}, nil
+}
+
+func (d *Driver) DeleteSnapshot(opt *pb.DeleteVolumeSnapshotOpts) error {
+	log.Infof("%v: trying delete snapshot...", DriverName)
+	poolId := opt.GetMetadata()["PoolId"]
+	err := d.client.DeleteSnapshot(poolId, opt)
+	if err != nil {
+		log.Errorf("%v: delete volume snapshot failed, volume snapshot id = %s , error: %v", DriverName, opt.GetId(), err)
+		return err
+	}
+	log.Infof("%v: remove volume snapshot success, volume snapshot id=%v", DriverName, opt.GetId())
+	return nil
+}
+
+func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
+	log.Infof("%v: listPools ...", DriverName)
+	var pols []*model.StoragePoolSpec
+	sp, err := d.client.ListStoragePools()
+	if err != nil {
+		return nil, err
+	}
+
+	c := d.conf
+
+	for _, pool := range sp {
+		for grpName, _ := range c.Pool {
+			if grpName == pool.Name && c.Pool[grpName].AvailabilityZone == pool.ArrayList[0].ArrayName {
+
+				pol := &model.StoragePoolSpec{
+					BaseModel: &model.BaseModel{
+						//				Id: pool.Id,
```

Comment:
Please remove these comments if not required

Reply:
I removed them...